### PR TITLE
feat(The New York Times): add features, edit code

### PR DESCRIPTION
**Pull request by 7loud**
**Describe the changes in this pull request:**

- Added "myaccount.nytimes.com"
- Added some features
- Updated code

<details>
<summary> Proof showing the creation/modification is working as expected </summary>
<br>

<hr>

https://github.com/PreMiD/Presences/pull/5632/commits/cee52fe389d9a843c7cb43b1db1e9bd2e5e69654

<hr>

https://github.com/PreMiD/Presences/pull/5632/commits/8aa7b9ced667918cac1116afad0ba026299bb5c8

<hr>

https://github.com/PreMiD/Presences/pull/5632/commits/52350f7d1835ae34977bfe70df91c4326695905a

<hr>

`John Smith uploaded the picture`

</details>

image

### CODE REVIEW

**Review by Bas950 — CHANGES_REQUESTED** [review]
use tabs

File: websites/T/The New York Times/presence.ts [hunk 1]
```diff
@@ -4,59 +4,217 @@ const presence = new Presence({
...
-		details = "Viewing an Article: ";
-		state = title.replace(" - The New York Times", "");
-	}
+	const { title } = document,
+		setting = {
+			privacy: await presence.getSetting("privacy"),
+			buttons: await presence.getSetting("buttons"),
+			podcastLogo: await presence.getSetting("podcastLogo"),
+			articleAuthor: await presence.getSetting("articleAuthor"),
+			moreDetails: await presence.getSetting("moreDetails")
```
Comment (Bas950) [inline websites/T/The New York Times/presence.ts:13]: cant you like promise.all this?

**Review by Bas950 — DISMISSED** [review]

File: websites/T/The New York Times/presence.ts (hunk 1 above)
Comment (EncryptedDev) [reply websites/T/The New York Times/presence.ts:13]: and add types

File: websites/T/The New York Times/presence.ts [hunk 2]
```diff
@@ -4,59 +4,217 @@ const presence = new Presence({
...
-		details = "Viewing a Video Section: ";
-		state = title.replace(" - The New York Times", "");
-	} else if (document.location.pathname.includes("/podcasts/the-daily/")) {
-		details = "Viewing a Podcast: ";
-		state = `The Daily: ${title.replace(" - The New York Times", "")}`;
-	} else {
-		details = "Viewing an Article: ";
-		state = title.replace(" - The New York Times", "");
-	}
+	const { title } = document,
```
Comment (Slowlife01) [inline websites/T/The New York Times/presence.ts:7]:
```suggestion
	const title = document.title.replace(" - The New York Times", ""),
```

File: websites/T/The New York Times/presence.ts [hunk 3]
```diff
@@ -4,59 +4,233 @@ const presence = new Presence({
...
+				);
+
+			presenceData.details = setting.privacy
+				? "Listening to a Podcast"
+				: "Listening to a Podcast:";
+			if (podcast && !setting.privacy) {
+				presenceData.state = `${podcast.textContent}: ${title.replace(
+					" - The New York Times",
+					""
+				)}`;
```
Comment (Slowlife01) [inline websites/T/The New York Times/presence.ts:83]:
```suggestion
				presenceData.state = `${podcast.textContent}: ${title}`;
```

**Review by Slowlife01 — APPROVED** [review]

**Review by EncryptedDev — APPROVED** [review]

**Pull request merged**